File: SoloLearn Codes/Name Buddy/NameBuddy.java
import java.util.Arrays;
import java.util.Scanner;

public class NameBuddy
{
    public static void main(String[] args) {
		Scanner input = new Scanner(System.in);	
		String[] names = input.nextLine().split(" ");	  
		String name = input.nextLine();
		System.out.println(Arrays
				.stream(names)
				.anyMatch(a -> a.charAt(0) == name.charAt(0)) ? "Compare notes" : "No such luck");
    }
}

File: SoloLearn Codes/Name Buddy/NameBuddy.cs
using System;
using System.Linq;

namespace SoloLearn
{
    class Program
    {
        static void Main(string[] args)
        {
            string[] g = Console.ReadLine().Split();
            string n = Console.ReadLine();
            Console.WriteLine(g.Any(m => m[0] == n[0]) ? "Compare notes" : "No such luck");
        }
    }
}